server/channels/app/user.go

```diff
@@ -1038,6 +1038,58 @@ func (a *App) userDeactivated(rctx request.CTX, userID string) *model.AppError {
 		rctx.Logger().Warn("unable to remove auth data by user id", mlog.Err(nErr))
 	}
 
+	if err = a.softDeleteUserDMChannels(rctx, userID, model.GetMillis()); err != nil {
+		rctx.Logger().Warn("Failed to archive DM channels for deactivated user", mlog.Err(err))
+	}
+
+	return nil
+}
+
+func (a *App) softDeleteUserDMChannels(rctx request.CTX, userID string, deleteTimestamp int64) *model.AppError {
+	fromID := ""
+	const pageSize = 200
+
+	for {
+		channels, err := a.Srv().Store().Channel().GetChannelsByUser(userID, false /* includeDeleted */, 0, pageSize, fromID)
+		if err != nil {
+			return model.NewAppError("softDeleteUserDMChannels", "app.channel.get_channels_by_user.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
+		}
+		if len(channels) == 0 {
+			break
+		}
```

Comment:
This is redundant as you are already checking the size of queried channels below: https://github.com/mattermost/mattermost/pull/33774/files#diff-03e2702288b2a1348d400efa6e5ac2dfc6748997f099f46a2366bc8a85efa357R1088